src/openrct2/object/ObjectList.h

```diff
@@ -19,7 +19,8 @@ class ObjectList
     std::vector<std::vector<ObjectEntryDescriptor>> _subLists;
 
 public:
-    void Add(const ObjectEntryDescriptor& entry);
+    ObjectEntryIndex Add(const ObjectEntryDescriptor& entry);
+    ObjectEntryIndex Add(ObjectType type, std::string_view identifier);
```

Comment:
Do we need the two functions or do we want people to construct a proper `ObjectEntryDescriptor` upon calling?